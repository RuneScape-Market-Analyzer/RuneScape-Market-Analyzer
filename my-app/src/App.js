import React, { useState, useEffect } from 'react';
import { Bar, Line } from 'react-chartjs-2';

import {
  Chart as ChartJS, CategoryScale, LinearScale, PointElement, LineElement, BarElement, Title, Tooltip, Legend
} from 'chart.js';

ChartJS.register(CategoryScale, LinearScale, PointElement, LineElement, BarElement, Title, Tooltip, Legend);

// Background Square (decorative - serves no actual function)
const backgroundSquarePlugin = {
  id: 'backgroundSquare',
  beforeDraw: (chart) => {
    const { ctx, chartArea } = chart;
    const { left, right, top, bottom } = chartArea;

    ctx.save();
    ctx.fillStyle = 'rgba(118, 120, 237, 0.2)';
    ctx.fillRect(left + 10, top + 10, right - left - 20, bottom - top - 20);
    ctx.restore();
  },
};

// Bar Chart Setup
const barChartOptions = {
  responsive: true,
  maintainAspectRatio: false,
  plugins: {
    legend: { display: true, position: 'top' },
    title: {
      display: true,
      text: 'Item Price Trends',
      font: { size: 20 },
      color: 'rgba(33, 33, 33, 1)',
    },
  },
  scales: {
    x: { grid: { color: 'rgba(61, 52, 139, 0.3)' } },
    y: { beginAtZero: true, grid: { color: 'rgba(61, 52, 139, 0.2)' } },
  },
};

// Line Chart SetUp
const lineChartOptions = {
  responsive: true,
  maintainAspectRatio: false,
  plugins: {
    legend: { display: true, position: 'top' },
    title: { display: false },
  },
  scales: {
    x: { grid: { color: 'rgba(61, 52, 139, 0.3)' } },
    y: { beginAtZero: true, grid: { color: 'rgba(61, 52, 139, 0.2)' } },
  },
};

// Manging states for Item IDs, Selected Item, and Chart Data
function App() {
  const [itemIDs, setItemIDs] = useState([]);
  const [selectedItem, setSelectedItem] = useState('');
  const [chartData, setChartData] = useState({
    labels: [],
    datasets: [{
      label: `Price for Item ${selectedItem || '...'}`,
      data: [],
      backgroundColor: 'rgba(118, 120, 237, 0.8)',
      borderColor: 'rgba(61, 52, 139, 1)',
      borderWidth: 2,
      pointBackgroundColor: 'rgba(118, 120, 237, 1)',
    }],
  });  

  // Get item IDs
  useEffect(() => {
    const fetchItemIDs = async () => {
      try {
        const response = await fetch('http://127.0.0.1:5000/items/ids');
        const data = await response.json();
        console.log("Fetched Item IDs:", data);

        if (Array.isArray(data) && data.length > 0) {
          setItemIDs(data);
          setSelectedItem(data[0]);
          fetchPriceData(data[0]);
        }
      } catch (error) {
        console.error("Error fetching item IDs:", error);
      }
    };

    fetchItemIDs();
  }, []);

  // Get price data for selected item
  const fetchPriceData = async (itemID) => {
    try {
      const response = await fetch(`http://127.0.0.1:5000/items/prices/${itemID}`);
      const data = await response.json();
      console.log("Fetched Price Data:", data);

      if (!Array.isArray(data) || data.length === 0 || !Array.isArray(data[0]) || data[0].length < 3) {
        throw new Error("Invalid data format received from backend");
      }

      const timestamps = data.map(entry => entry[1]);
      const prices = data.map(entry => entry[2]);

      setChartData({
        labels: timestamps,
        datasets: [{
          label: `Price for Item ${itemID}`,
          data: prices,
          backgroundColor: 'rgba(118, 120, 237, 0.8)',
          borderColor: 'rgba(61, 52, 139, 1)',
          borderWidth: 2,
          pointBackgroundColor: 'rgba(118, 120, 237, 1)',
        }],
      });

    } catch (error) {
      console.error("Error fetching data:", error);
      alert("Failed to fetch price data. Check backend and item ID.");
    }
  };

  const [search, setSearch] = useState('');

  const filteredItems = itemIDs.filter(id => id.toString().includes(search));

  return (
    <div style={{ width: '800px', margin: 'auto', textAlign: 'center' }}>
      <h2>RuneScape Market Trends</h2>
      
      {/* Search and Item Menu */}
      <div style={{ marginBottom: '20px', display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
        <label style={{ marginBottom: '5px', fontSize: '16px', fontWeight: 'bold' }}>Select an Item:</label>

        {/* Search Bar */}
        <input
          type="text"
          placeholder="Search item..."
          value={search}
          onChange={(e) => setSearch(e.target.value)}
          style={{
            width: '185px',
            padding: '8px',
            marginBottom: '5px',
            fontSize: '14px',
            border: '1px solid #ccc',
            borderRadius: '5px',
          }}
        />

        {/* Scroll Item Menu */}
        <select 
          value={selectedItem} 
          onChange={(e) => {
            setSelectedItem(e.target.value);
            fetchPriceData(e.target.value);
          }}
          style={{
            padding: '9px',
            fontSize: '16px',
            width: '200px',
            maxHeight: '73px',
            overflowY: 'auto',
          }}
          size={3}
        >
          {filteredItems.length > 0 ? (
            filteredItems.map(id => (
              <option key={id} value={id}>
                {id}
              </option>
            ))
          ) : (
            <option>No matching items</option>
          )}
        </select>
      </div>

      {/* Bar Chart */}
      <div style={{ width: '100%', height: '500px' }}>
        <Bar 
          data={{
            ...chartData,
            datasets: [{
              ...chartData.datasets[0],
              barThickness: 6,
            }]
          }} 
          options={{
            ...barChartOptions, 
            maintainAspectRatio: true,
          }} 
          plugins={[backgroundSquarePlugin]} 
        />
      </div>

      {/* Line Chart */}
      <div style={{ width: '100%', height: '500px' }}>
        <Line 
          data={chartData} 
          options={{
            ...lineChartOptions, 
            maintainAspectRatio: true,
          }} 
          plugins={[backgroundSquarePlugin]} 
        />
      </div>
    </div>
  );
}

export default App;