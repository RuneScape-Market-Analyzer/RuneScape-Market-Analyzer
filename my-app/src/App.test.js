// App.test.js
import { render, screen } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom';

describe('Basic Component Checks', () => {
  test('renders at least one header', () => {
    render(<App />);
    // Check for existence rather than specific content
    expect(screen.getAllByRole('heading').length).toBeGreaterThan(0);
  });

  test('has navigation links', () => {
    render(<App />);
    // Check for presence without worrying about duplicates
    expect(screen.getAllByText(/Home/i).length).toBeGreaterThan(0);
    expect(screen.getAllByText(/News/i).length).toBeGreaterThan(0);
    expect(screen.getAllByText(/FAQ/i).length).toBeGreaterThan(0);
  });

  test('displays news content', () => {
    render(<App />);
    // Generic checks for news section elements
    expect(screen.getByText(/Latest News/i)).toBeInTheDocument();
    expect(screen.getAllByText(/Update/i).length).toBeGreaterThan(0);
  });

  test('has footer content', () => {
    render(<App />);
    expect(screen.getByText(/RuneScape Fan Site/i)).toBeInTheDocument();
  });
});

describe('Presence Checks', () => {
  test('has interactive elements', () => {
    render(<App />);
    expect(screen.getAllByRole('button').length).toBeGreaterThan(0);
    expect(screen.getAllByRole('img').length).toBeGreaterThan(0);
  });
});

test('matches snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});