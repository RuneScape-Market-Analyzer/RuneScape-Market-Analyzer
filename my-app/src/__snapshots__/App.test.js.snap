// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches generic snapshot 1`] = `
<div>
  <div
    style="min-height: 100vh; color: white;"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        <img
          alt="Logo"
          src="RuneScape.png"
        />
      </div>
      <ul
        class="nav-links"
      >
        <li>
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/trade"
          >
            Trade
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/news"
          >
            News
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/faq"
          >
            FAQ
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
    <div
      style="flex: 1;"
    >
      <div
        style="text-align: center; margin-top: 50px;"
      >
        <div
          class="trading-container"
        >
          <div
            class="top-section"
          >
            <div
              class="left-content"
            >
              <div
                class="left-content-inner"
              >
                <h1>
                  Stay Updated with Your Gaming Stats
                </h1>
                <p>
                  Discover your RuneScape trading graph, and price analysis
                </p>
                <button
                  class="navigate-button"
                >
                  Explore More
                </button>
              </div>
            </div>
            <div
              class="right-content"
            >
              <div
                class="top-movers new-layout"
              >
                <div
                  class="top-header"
                >
                  <div
                    class="header-text"
                  >
                    <h2
                      style="opacity: 0; transform: translateY(6px) translateZ(0);"
                    >
                      Today's Market Trends
                    </h2>
                    <p
                      class="subheader"
                      style="opacity: 0; transform: translateY(4px) translateZ(0);"
                    >
                      Showing most traded items by volume
                    </p>
                  </div>
                  <div
                    class="toggle-buttons"
                  >
                    <button
                      class="active volume-active"
                    >
                      Volume
                    </button>
                    <button
                      class=""
                    >
                      Gainers
                    </button>
                    <button
                      class=""
                    >
                      Decliners
                    </button>
                  </div>
                </div>
                <div
                  class="mover-list full"
                >
                  <div
                    class="table-header"
                  >
                    <div
                      class="header-icon"
                    />
                    <div
                      class="header-name"
                    >
                      Item
                    </div>
                    <div
                      class="header-price"
                    >
                      Price
                    </div>
                    <div
                      class="header-change"
                    >
                      Volume
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h1>
            Most Traded Item Of Today
          </h1>
          <div
            class="most-traded-item"
          >
            Most Traded Item
          </div>
        </div>
        <div
          class="news-container"
        >
          <div
            class="news-container"
          >
            <div
              class="news-title"
            >
              <h1
                class="global-title"
              >
                RuneScape News & Updates
              </h1>
            </div>
            <div
              class="featured-news"
            >
              <div
                class="featured-text"
              >
                <span
                  class="news-tag"
                >
                  Featured Update
                </span>
                <h2
                  class="news-heading"
                >
                  Return to the Desert: Pharaoh's Folly
                </h2>
                <p
                  class="news-date"
                >
                  April 5, 2024
                </p>
                <p
                  class="news-summary"
                >
                  Continuing the story from Beneath Scabaras’ Sands, you’ll journey back to Menaphos, team up with Ozan and Leela to uncover Pharaoh Osman’s betrayal.
                </p>
              </div>
              <div
                class="featured-image-wrapper"
              >
                <img
                  alt="Return to the Desert: Pharaoh's Folly"
                  class="featured-image"
                  src="https://cdn.runescape.com/assets/img/external/news/2025/04/pfkey.jpg"
                />
              </div>
            </div>
            <div
              class="news-section-header"
            >
              <h2
                class="news-section-title"
              >
                Latest News
              </h2>
              <button
                class="news-read-more-link"
              >
                Read More 
                <span
                  class="arrow"
                >
                  →
                </span>
              </button>
            </div>
            <div
              class="news-grid"
            >
              <article
                class="news-card"
              >
                <img
                  alt="New Elder God Wars Dungeon Update"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/The_Croesus_Front_Survival_Guide_(1)_update_image.jpg/1160px-The_Croesus_Front_Survival_Guide_(1)_update_image.jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Game Update
                  </span>
                  <h2
                    class="news-heading"
                  >
                    New Elder God Wars Dungeon Update
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 14, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    The final front of the Elder God Wars Dungeon arrives with challenging new bosses and powerful rewards...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Double XP Weekend Announced"
                  class="news-image"
                  src="https://external-preview.redd.it/Q6md3IHfYgDkCs32DHnAVj5-Fj_XGxv_JIFlTSjSSvQ.jpg?auto=webp&s=515a653678ce9bab4a4b510b5cb85d1516c1ab27"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Event
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Double XP Weekend Announced
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 11, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Prepare your training plans! The next Double XP Weekend will run from March 24th to 28th...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Wilderness Rework Proposal"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg/640px-Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Developer Blog
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Wilderness Rework Proposal
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 9, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Jagex releases proposed changes to Wilderness PvP mechanics. Player feedback needed...
                  </p>
                </div>
              </article>
            </div>
          </div>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
        <div
          class="faq-container"
        >
          <h1>
            Have Questions?
          </h1>
          <p>
            Check out our FAQ section!
          </p>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
      </div>
    </div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-content"
      >
        <p
          class="footer-text"
        >
          © 
          2025
           RuneScape Fan Site. All rights reserved.
        </p>
        <div
          class="footer-links"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/trade"
          >
            Trade
          </a>
          <a
            href="/news"
          >
            News
          </a>
          <a
            href="/faq"
          >
            FAQ
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div
    style="min-height: 100vh; color: white;"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        <img
          alt="Logo"
          src="RuneScape.png"
        />
      </div>
      <ul
        class="nav-links"
      >
        <li>
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/trade"
          >
            Trade
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/news"
          >
            News
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/faq"
          >
            FAQ
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
    <div
      style="flex: 1;"
    >
      <div
        style="text-align: center; margin-top: 50px;"
      >
        <div
          class="trading-container"
        >
          <div
            class="top-section"
          >
            <div
              class="left-content"
            >
              <div
                class="left-content-inner"
              >
                <h1>
                  Stay Updated with Your Gaming Stats
                </h1>
                <p>
                  Discover your RuneScape trading graph, and price analysis
                </p>
                <button
                  class="navigate-button"
                >
                  Explore More
                </button>
              </div>
            </div>
            <div
              class="right-content"
            >
              <div
                class="top-movers new-layout"
              >
                <div
                  class="top-header"
                >
                  <div
                    class="header-text"
                  >
                    <h2
                      style="opacity: 0; transform: translateY(6px) translateZ(0);"
                    >
                      Today's Market Trends
                    </h2>
                    <p
                      class="subheader"
                      style="opacity: 0; transform: translateY(4px) translateZ(0);"
                    >
                      Showing most traded items by volume
                    </p>
                  </div>
                  <div
                    class="toggle-buttons"
                  >
                    <button
                      class="active volume-active"
                    >
                      Volume
                    </button>
                    <button
                      class=""
                    >
                      Gainers
                    </button>
                    <button
                      class=""
                    >
                      Decliners
                    </button>
                  </div>
                </div>
                <div
                  class="mover-list full"
                >
                  <div
                    class="table-header"
                  >
                    <div
                      class="header-icon"
                    />
                    <div
                      class="header-name"
                    >
                      Item
                    </div>
                    <div
                      class="header-price"
                    >
                      Price
                    </div>
                    <div
                      class="header-change"
                    >
                      Volume
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h1>
            Most Traded Item Of Today
          </h1>
          <div
            class="most-traded-item"
          >
            Most Traded Item
          </div>
        </div>
        <div
          class="news-container"
        >
          <div
            class="news-container"
          >
            <div
              class="news-title"
            >
              <h1
                class="global-title"
              >
                RuneScape News & Updates
              </h1>
            </div>
            <div
              class="featured-news"
            >
              <div
                class="featured-text"
              >
                <span
                  class="news-tag"
                >
                  Featured Update
                </span>
                <h2
                  class="news-heading"
                >
                  Return to the Desert: Pharaoh's Folly
                </h2>
                <p
                  class="news-date"
                >
                  April 5, 2024
                </p>
                <p
                  class="news-summary"
                >
                  Continuing the story from Beneath Scabaras’ Sands, you’ll journey back to Menaphos, team up with Ozan and Leela to uncover Pharaoh Osman’s betrayal.
                </p>
              </div>
              <div
                class="featured-image-wrapper"
              >
                <img
                  alt="Return to the Desert: Pharaoh's Folly"
                  class="featured-image"
                  src="https://cdn.runescape.com/assets/img/external/news/2025/04/pfkey.jpg"
                />
              </div>
            </div>
            <div
              class="news-section-header"
            >
              <h2
                class="news-section-title"
              >
                Latest News
              </h2>
              <button
                class="news-read-more-link"
              >
                Read More 
                <span
                  class="arrow"
                >
                  →
                </span>
              </button>
            </div>
            <div
              class="news-grid"
            >
              <article
                class="news-card"
              >
                <img
                  alt="New Elder God Wars Dungeon Update"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/The_Croesus_Front_Survival_Guide_(1)_update_image.jpg/1160px-The_Croesus_Front_Survival_Guide_(1)_update_image.jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Game Update
                  </span>
                  <h2
                    class="news-heading"
                  >
                    New Elder God Wars Dungeon Update
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 14, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    The final front of the Elder God Wars Dungeon arrives with challenging new bosses and powerful rewards...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Double XP Weekend Announced"
                  class="news-image"
                  src="https://external-preview.redd.it/Q6md3IHfYgDkCs32DHnAVj5-Fj_XGxv_JIFlTSjSSvQ.jpg?auto=webp&s=515a653678ce9bab4a4b510b5cb85d1516c1ab27"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Event
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Double XP Weekend Announced
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 11, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Prepare your training plans! The next Double XP Weekend will run from March 24th to 28th...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Wilderness Rework Proposal"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg/640px-Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Developer Blog
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Wilderness Rework Proposal
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 9, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Jagex releases proposed changes to Wilderness PvP mechanics. Player feedback needed...
                  </p>
                </div>
              </article>
            </div>
          </div>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
        <div
          class="faq-container"
        >
          <h1>
            Have Questions?
          </h1>
          <p>
            Check out our FAQ section!
          </p>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
      </div>
    </div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-content"
      >
        <p
          class="footer-text"
        >
          © 
          2025
           RuneScape Fan Site. All rights reserved.
        </p>
        <div
          class="footer-links"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/trade"
          >
            Trade
          </a>
          <a
            href="/news"
          >
            News
          </a>
          <a
            href="/faq"
          >
            FAQ
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`renders homepage 1`] = `
<div>
  <div
    style="min-height: 100vh; color: white;"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        <img
          alt="Logo"
          src="RuneScape.png"
        />
      </div>
      <ul
        class="nav-links"
      >
        <li>
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/trade"
          >
            Trade
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/news"
          >
            News
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/faq"
          >
            FAQ
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
    <div
      style="flex: 1;"
    >
      <div
        style="text-align: center; margin-top: 50px;"
      >
        <div
          class="trading-container"
        >
          <div
            class="top-section"
          >
            <div
              class="left-content"
            >
              <div
                class="left-content-inner"
              >
                <h1>
                  Stay Updated with Your Gaming Stats
                </h1>
                <p>
                  Discover your RuneScape trading graph, and price analysis
                </p>
                <button
                  class="navigate-button"
                >
                  Explore More
                </button>
              </div>
            </div>
            <div
              class="right-content"
            >
              <div
                class="top-movers new-layout"
              >
                <div
                  class="top-header"
                >
                  <div
                    class="header-text"
                  >
                    <h2
                      style="opacity: 0; transform: translateY(6px) translateZ(0);"
                    >
                      Today's Market Trends
                    </h2>
                    <p
                      class="subheader"
                      style="opacity: 0; transform: translateY(4px) translateZ(0);"
                    >
                      Showing most traded items by volume
                    </p>
                  </div>
                  <div
                    class="toggle-buttons"
                  >
                    <button
                      class="active volume-active"
                    >
                      Volume
                    </button>
                    <button
                      class=""
                    >
                      Gainers
                    </button>
                    <button
                      class=""
                    >
                      Decliners
                    </button>
                  </div>
                </div>
                <div
                  class="mover-list full"
                >
                  <div
                    class="table-header"
                  >
                    <div
                      class="header-icon"
                    />
                    <div
                      class="header-name"
                    >
                      Item
                    </div>
                    <div
                      class="header-price"
                    >
                      Price
                    </div>
                    <div
                      class="header-change"
                    >
                      Volume
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h1>
            Most Traded Item Of Today
          </h1>
          <div
            class="most-traded-item"
          >
            Most Traded Item
          </div>
        </div>
        <div
          class="news-container"
        >
          <div
            class="news-container"
          >
            <div
              class="news-title"
            >
              <h1
                class="global-title"
              >
                RuneScape News & Updates
              </h1>
            </div>
            <div
              class="featured-news"
            >
              <div
                class="featured-text"
              >
                <span
                  class="news-tag"
                >
                  Featured Update
                </span>
                <h2
                  class="news-heading"
                >
                  Return to the Desert: Pharaoh's Folly
                </h2>
                <p
                  class="news-date"
                >
                  April 5, 2024
                </p>
                <p
                  class="news-summary"
                >
                  Continuing the story from Beneath Scabaras’ Sands, you’ll journey back to Menaphos, team up with Ozan and Leela to uncover Pharaoh Osman’s betrayal.
                </p>
              </div>
              <div
                class="featured-image-wrapper"
              >
                <img
                  alt="Return to the Desert: Pharaoh's Folly"
                  class="featured-image"
                  src="https://cdn.runescape.com/assets/img/external/news/2025/04/pfkey.jpg"
                />
              </div>
            </div>
            <div
              class="news-section-header"
            >
              <h2
                class="news-section-title"
              >
                Latest News
              </h2>
              <button
                class="news-read-more-link"
              >
                Read More 
                <span
                  class="arrow"
                >
                  →
                </span>
              </button>
            </div>
            <div
              class="news-grid"
            >
              <article
                class="news-card"
              >
                <img
                  alt="New Elder God Wars Dungeon Update"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/The_Croesus_Front_Survival_Guide_(1)_update_image.jpg/1160px-The_Croesus_Front_Survival_Guide_(1)_update_image.jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Game Update
                  </span>
                  <h2
                    class="news-heading"
                  >
                    New Elder God Wars Dungeon Update
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 14, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    The final front of the Elder God Wars Dungeon arrives with challenging new bosses and powerful rewards...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Double XP Weekend Announced"
                  class="news-image"
                  src="https://external-preview.redd.it/Q6md3IHfYgDkCs32DHnAVj5-Fj_XGxv_JIFlTSjSSvQ.jpg?auto=webp&s=515a653678ce9bab4a4b510b5cb85d1516c1ab27"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Event
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Double XP Weekend Announced
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 11, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Prepare your training plans! The next Double XP Weekend will run from March 24th to 28th...
                  </p>
                </div>
              </article>
              <article
                class="news-card"
              >
                <img
                  alt="Wilderness Rework Proposal"
                  class="news-image"
                  src="https://runescape.wiki/images/thumb/Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg/640px-Wilderness_rework_with_opt-in_PvP_out_now!_-_The_Wilderness_Reborn_(RuneScape).jpg"
                />
                <div
                  class="news-body"
                >
                  <span
                    class="news-tag"
                  >
                    Developer Blog
                  </span>
                  <h2
                    class="news-heading"
                  >
                    Wilderness Rework Proposal
                  </h2>
                  <p
                    class="news-date"
                  >
                    March 9, 2024
                  </p>
                  <p
                    class="news-summary"
                  >
                    Jagex releases proposed changes to Wilderness PvP mechanics. Player feedback needed...
                  </p>
                </div>
              </article>
            </div>
          </div>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
        <div
          class="faq-container"
        >
          <h1>
            Have Questions?
          </h1>
          <p>
            Check out our FAQ section!
          </p>
          <button
            class="navigate-button"
          >
            View All
          </button>
        </div>
      </div>
    </div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-content"
      >
        <p
          class="footer-text"
        >
          © 
          2025
           RuneScape Fan Site. All rights reserved.
        </p>
        <div
          class="footer-links"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/trade"
          >
            Trade
          </a>
          <a
            href="/news"
          >
            News
          </a>
          <a
            href="/faq"
          >
            FAQ
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
